public/src/components/epicTests/epicTestsForm.tsx

```diff
@@ -138,26 +165,87 @@ class EpicTestsForm extends React.Component<EpicTestFormProps, EpicTestsFormStat
       })
   }
 
+  requestEpicTestsLock = () => {
+    requestLock(FrontendSettingsType.epicTests).then(response =>
+      response.ok ? this.fetchStateFromServer() : alert("Error - can't request lock!")
+    );
+  }
+
+  requestTakeControl = () => {
+    requestTakeControl(FrontendSettingsType.epicTests).then(response =>
+      response.ok ? this.fetchStateFromServer() : alert("Error - can't take back control!")
+    );
+  }
+
+  renderButtonsBar = () => {
+    const {classes} = this.props;
+    if (!this.state.editMode) {
+      if (this.state.lockStatus.locked) {
+        const friendlyName: string | undefined = this.state.lockStatus.email && this.makeFriendlyName(this.state.lockStatus.email);
+        const friendlyTimestamp: string | undefined = this.state.lockStatus.timestamp && this.makeFriendlyDate(this.state.lockStatus.timestamp);
+        return (
+          <>
+            <Typography className={classes.warning}>File locked for editing by {friendlyName} (<a href={"mailto:" + this.state.lockStatus.email}>{this.state.lockStatus.email}</a>) at {friendlyTimestamp}</Typography>
+            <div className={classes.buttons}>
+              <ButtonWithConfirmationPopup
+                buttonText="Take control"
+                confirmationText={`Are you sure? Please tell ${friendlyName} that their unpublished changes will be lost.`}
+                onConfirm={this.requestTakeControl}
+                icon={<LockOpenIcon />}
+              />
+            </div>
+          </>
+        );
+      } else {
+        return (
+          <Button
+            onClick={this.requestEpicTestsLock}
+            variant="contained" color="primary"
+          >Edit</Button>
+        )
+      }
+    } else {
+      return (
+        <div>
+          <ButtonWithConfirmationPopup
+          buttonText="Publish"
+          confirmationText="Are you sure? This will replace all live tests!"
+          onConfirm={this.save}
+          icon={<CloudUploadIcon />}
+        />
+        <ButtonWithConfirmationPopup
+        buttonText="Cancel"
+        confirmationText="Are you sure? All unpublished data will be lost!"
+        onConfirm={() => this.cancel()}
+        icon={<RefreshIcon />}
+        />
+        </div>
+        )
+    }
+  }
+
+  makeFriendlyDate = (timestamp: string): string => {
+    const year = timestamp.slice(0, 4);
+    const month = timestamp.slice(5,7);
+    const day = timestamp.slice(8,10);
+    const time = timestamp.slice(11,19);
+    return `${time} on ${day}/${month}/${year}`;
+  }
+
+  makeFriendlyName = (email: string): string => {
```

Comment:
should this include the second name?

Reply:
I wasn't sure - was trying to keep it friendly as there is such a small number of users - what do you think?

Reply:
I think there's a good chance of having 2 people with the same first name!

Reply:
True - ok will add last name too.